vowpalwabbit/cb_sample.cc

```diff
@@ -78,14 +81,34 @@ struct cb_sample_data
       {
         _random_state->get_and_update_random();
       }
+
     }
 
     auto result = exploration::swap_chosen(action_scores.begin(), action_scores.end(), chosen_action);
     assert(result == S_EXPLORATION_OK);
 
+    VW_DBG(examples) << "cb " << cb_decision_to_string(examples[0]->pred.a_s) <<
+                    " rnd:" << _random_state->get_current_state() << std::endl;
+
     _UNUSED(result);
   }
 
+  std::string cb_decision_to_string(const ACTION_SCORE::action_scores& action_scores)
```

Comment:
You can just add an overload for `operator<<` in v_array.h to handle this in a uniform way that will be discoverable and usable elsewhere in the codebase

Reply:
The following does not seem appropriate in varray.h since it is a specific interpretation of what the action, score collection means here.

        ostrm << "chosen[" << a_s.action << "," << a_s.score << "] [ "

Reply:
Yes, I agree. But you can reasonably provide a `operator<<` impl for `action_score` and then the existing `operator<<` for `v_array<T>` would utilize this when printing.

Reply:
Done.

Reply:
@rajan-chari My comment was that the entire `cb_decision_to_string` function is not necessary. Now that you have an `operator<<` for `action_score` this entire function can be replaced with simply passing a `v_array<T>` to `operator<<`